### DISCUSSION TITLE
CSV Reader IndexOutOfBoundsException

### DISCUSSION THREAD

**Discussion by anand-srinivaas** [jtablesaw/tablesaw#996] [Q&A] [answered]
Hi,

Using version : 0.38.1

In a csv file, if one of the records is incorrect carrying additional delimiter at the end [ extra comma ] , then ColumnTypeDetector.detectColumnTypes is throwing IndexOutOfBoundsException without adding any additional information like row number. 

Whereas if there is a comma missing in any of the records inside a csv file, properly throwing below exception tech.tablesaw.io.AddCellToColumnException: Error while adding cell from row 2 and column email(position:0): Row number 2 contains 4 columns. 5 expected.

I believe, it would be useful if the exception carries row number in either cases. That way, it will allow user an option to identify the problematic row and also to skip the same if required - Just a thought.

Thanks,
Anand

**Comment by lwhite1** [comment, marked as answer]
I opened an issue for this problem #998

**Reply by lwhite1** [reply]
Fixed in 0.42.0-SNAPSHOT

**Reply by anand-srinivaas** [reply]
Thanks @lwhite1  !
